api/model/src/main/java/org/projectnessie/model/Namespace.java

```diff
@@ -48,6 +49,18 @@ public abstract class Namespace extends Content {
   static final String ERROR_MSG_TEMPLATE =
       "'%s' is not a valid namespace identifier (should not end with '.')";
 
+  /** This separate static class is needed to prevent class loader deadlocks. */
+  public static final class Empty {
+    private Empty() {}
+
+    public static final Namespace EMPTY_NAMESPACE = builder().build();
+  }
+
+  /**
+   * Refactor all code references to this constant to use {@link Empty#EMPTY_NAMESPACE}, there's a
+   * non-zero risk of causing a class loader deadlock.
+   */
+  @Deprecated
   public static final Namespace EMPTY = builder().elements(Collections.emptyList()).build();
```

Comment:
One more nittish comment, but this is public API, so I hope it might be cleaner for outside callers:

We could probably have in `Namespace`:
```
public static Namespace emptyNamespace() {
   return Empty.EMPTY_NAMESPACE;
}
```
... then `Empty` can be `private`... WDYT?

Reply:
Nah - let's keep it as is